eclipse-temurin-17-maven-4/Dockerfile

```diff
@@ -1,6 +1,6 @@
 FROM eclipse-temurin:17-jdk AS builder
 
-ARG MAVEN_VERSION=4.0.0-rc-2
+ARG MAVEN_VERSION=4.0.0-rc-3
 ARG USER_HOME_DIR="/root"
 ARG SHA=3875f0387334d1a3941f344f654be30d9687c997faa599d7144f312d0a14d3d23fe35828223f93a57bf2682f8e6e2ad601a9c3dc28a8488acee13d1ad4e53dfe
```

Comment:
this SHA should be updated 
the windows shas for maven 3 should not be updated in this PR
https://github.com/carlossg/docker-maven?tab=readme-ov-file#updating-maven-version